src/dir.targets

```diff
@@ -157,9 +154,39 @@
     <DebugEngines>{2E36F1D4-B23C-435D-AB41-18E608940038}</DebugEngines>
   </PropertyGroup>
 
+  <Target Name="CheckTestCategories"
+          AfterTargets="Test"
+          Condition="'$(IsTestProject)'=='True' And '$(SkipTests)'!='True'">
+    <PropertyGroup>
+      <!-- By default project has inner- and outer- loop tests. -->
+      <TestCategories Condition="'$(TestCategories)'==''">InnerLoop;OuterLoop</TestCategories>
+    </PropertyGroup>
+    <ItemGroup>
+      <!-- Split semicolon separated list -->
+      <TestCategoriesItems Include="$(TestCategories)" />
+      
+      <!-- If category not found then list will be empty -->
+      <InnerLoop Include="@(TestCategoriesItems)" Condition="'%(TestCategoriesItems.Identity)'=='InnerLoop'" />
+      <OuterLoop Include="@(TestCategoriesItems)" Condition="'%(TestCategoriesItems.Identity)'=='OuterLoop'" />
+    </ItemGroup>
+    <PropertyGroup>
+      <!-- If list not empty then category found -->
+      <IsInnerLoop Condition="'@(InnerLoop)'!=''">True</IsInnerLoop>
+      <IsOuterLoop Condition="'@(OuterLoop)'!=''">True</IsOuterLoop>
+
+      <!-- What categories of tests do we want to run by default  -->
+      <RunInnerLoop Condition="'$(RunInnerLoop)'==''">True</RunInnerLoop>
+      
+      <!-- Run tests only if we want to run tests of the given category and the test has this category -->
+      <RunTestsForProject>False</RunTestsForProject>
+      <RunTestsForProject Condition="'$(IsInnerLoop)'=='True' And '$(RunInnerLoop)'=='True'">True</RunTestsForProject>
+      <RunTestsForProject Condition="'$(IsOuterLoop)'=='True' And '$(RunOuterLoop)'=='True'">True</RunTestsForProject>
```

Comment:
I think what Wes is saying is that it would be nice if $(TestCategories) were the thing that controls the options passed to Xunit, instead of having individual properties like $(IsInnerLoop) for each category.  That way if we add a new category it just works.